### PR TITLE
Improve helper-fixtures

### PR DESCRIPTION
**Pull request by JLHwung**
| Q                        | A 
| ------------------------ | ---
| License                  | MIT

- Exclude tests for different major versions. For example, the `BABEL_8_BREAKING: true` tests will be completely ignored because they are never to be re-enabled. However, we can always investigate whether we should enable the skipped tests again.
- Reduce test files loading I/O, instead of checking every possible combinations of filename and supported extensions, we list the files in a test suite a matching them against the basename.

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/15890"></a>

### CODE REVIEW

File: packages/babel-helper-fixtures/src/index.ts [hunk 1]
```diff
@@ -120,57 +124,97 @@ function pushTask(
...
-  if (taskDirStats.isFile()) {
+    actualLoc ??= taskDir + "/input.js";
+    execLoc ??= taskDir + "/exec.js";
+    expectLoc ??= taskDir + "/output.js";
+
+    actualLocAlias =
+      suiteName + "/" + taskName + "/" + path.basename(actualLoc);
+    expectLocAlias =
+      suiteName + "/" + taskName + "/" + path.basename(actualLoc);
+    execLocAlias = suiteName + "/" + taskName + "/" + path.basename(actualLoc);
```
Comment (liuxingbaoyu) [inline packages/babel-helper-fixtures/src/index.ts:192]: I know they come from the original code, we can simplify it by the way. :)

**Review by liuxingbaoyu — APPROVED** [review]

**Review by nicolo-ribaudo — APPROVED** [review]
Does this also have a noticeable difference in the time to execute our tests?

File: packages/babel-helper-fixtures/src/index.ts (hunk 1 above)
Comment (JLHwung) [reply packages/babel-helper-fixtures/src/index.ts]: Good catch. I think they should be different.

**Comment by JLHwung** [comment]
> Does this also have a noticeable difference in the time to execute our tests?

Not as I can observe. The executing time remains around 23 seconds on my machine.

**Pull request merged**